src/cmds/qstat.c

```diff
@@ -3179,11 +3174,7 @@ qstat -B [-f] [-F format] [-D delim] [ server_name... ]\n";
 	 */
 	if (any_failed == PBSE_JOBHISTNOTSET)
 		any_failed = 0;
-#ifdef WIN32
-	Sleep(200);
-#else
-	usleep(200000);
-#endif
+	create_query_file();
```

Comment:
I would call this from check_last_query() if the file isn't present and make create_query_file() a static or eliminate it altogether. It doesn't need to be exposed to the clients themselves.

Reply:
@mike0042 i think we need to do the "touch" (or update timstamp) activity at the very exit of the program, else we will need to account for the varying time that the program might spend from the start (where check_last_query()) was called. In case of some qstat -f's it could be in the range of tens of seconds, making out check for the minimum delay bad or ineffective.

I suggest we register an atexit() handler that automatically calls the "create_query_file()" function internally. This way we do not have to cover all the exit paths of the client program. Also, the create_query_file() function does not have to be exposed to the client.